test/e2e/authorizer/authorizer_test.go

```diff
@@ -0,0 +1,249 @@
+		// obj, gvk, err := decoder.Decode(raw, nil, nil)
+		_, gvk, err := decoder.Decode(doc, nil, u)
+		if err != nil {
+			return fmt.Errorf("error decoding %s: %w", filename, err)
+		}
+
+		resource := gvk.Kind
+		resource = strings.ToLower(resource)
+		resource += "s"
+
+		gvr := schema.GroupVersionResource{
+			Group:    gvk.Group,
+			Version:  gvk.Version,
+			Resource: resource,
+		}
+		resourceClient := client.Resource(gvr)
+
+		if _, err := resourceClient.Create(ctx, u, metav1.CreateOptions{}); err != nil {
+			return fmt.Errorf("error creating %s: %w", filename, err)
+		}
+	}
+
+	return nil
+}
+
+type clients struct {
+	KubeClient kubernetes.Interface
+	KcpClient  kcp.Interface
+	Dynamic    dynamic.Interface
+}
+
+func newUserClient(t *testing.T, username, orgWorkspace, workspace string, cfg *rest.Config) *clients {
+	cfgCopy := rest.CopyConfig(cfg)
+	cfgCopy.Host = cfg.Host + "/clusters/" + orgWorkspace + "_" + workspace
+	cfgCopy.BearerToken = username + "-token"
+	kubeClient, err := kubernetes.NewForConfig(cfgCopy)
+	require.NoError(t, err)
+	kcpClient, err := kcp.NewForConfig(cfgCopy)
+	require.NoError(t, err)
+	dynamicClient, err := dynamic.NewForConfig(cfgCopy)
+	require.NoError(t, err)
+	return &clients{
+		KubeClient: kubeClient,
+		KcpClient:  kcpClient,
+		Dynamic:    dynamicClient,
+	}
+}
+
+func TestAuthorizer(t *testing.T) {
+	usersKCPArgs, err := framework.Users([]framework.User{
+		{
+			Name:   "user-1",
+			UID:    "1111-1111-1111-1111",
+			Token:  "user-1-token",
+			Groups: []string{"team-1"},
+		},
+		{
+			Name:   "user-2",
+			UID:    "1111-1111-1111-1111",
+			Token:  "user-2-token",
+			Groups: []string{"team-2"},
+		},
+		{
+			Name:   "user-3",
+			UID:    "1111-1111-1111-1111",
+			Token:  "user-3-token",
+			Groups: []string{"team-3"},
+		},
+	}).ArgsForKCP(t)
+	require.NoError(t, err)
+
+	f := framework.NewKCPFixture(framework.KcpConfig{
+		Name: "main",
+		Args: usersKCPArgs,
+	})
+	teardown := f.SetUp(t)
+	defer teardown()
+
+	ctx := context.Background()
+	if deadline, ok := t.Deadline(); ok {
+		withDeadline, cancel := context.WithDeadline(ctx, deadline)
+		t.Cleanup(cancel)
+		ctx = withDeadline
+	}
+	require.Equalf(t, len(f.Servers), 1, "incorrect number of servers")
+
+	server := f.Servers["main"]
+	kcpCfg, err := server.Config()
+	require.NoError(t, err)
+	clusterName, err := framework.DetectClusterName(kcpCfg, ctx, "workspaces.tenancy.kcp.dev")
+	require.NoError(t, err)
+	kubeClients, err := kubernetes.NewClusterForConfig(kcpCfg)
+	require.NoError(t, err)
+	kcpClients, err := kcp.NewClusterForConfig(kcpCfg)
+	require.NoError(t, err)
+	kcpClient := kcpClients.Cluster(clusterName)
+	require.NoError(t, err)
+	dynamicClient, err := dynamic.NewForConfig(kcpCfg)
+	require.NoError(t, err)
+
+	scheme := runtime.NewScheme()
+	err = rbacv1.AddToScheme(scheme)
+	require.NoError(t, err)
+	err = tenancyv1alpha1.AddToScheme(scheme)
+	require.NoError(t, err)
+	codecs := serializer.NewCodecFactory(scheme)
```

Comment:
I think you can delete this and pass `unstructured.UnstructuredJSONScheme` below

Reply:
hm, i need to recheck, but changing to this one fails e2e locally 🤔

Reply:
we can omit scheme registration though and reference `k8s.io/client-go/kubernetes/scheme#Scheme` directly, as types register here, for us in https://github.com/kcp-dev/kcp/blob/5887e9b6d59dc5fcbc75fb22f964cfc45ba43a5a/test/e2e/framework/util.go#L96-L98.

(sideeffects, sideeffects, sideffects, everywhere 😬 )